demisto_sdk/commands/common/tools.py

```diff
@@ -482,14 +482,70 @@ def get_local_remote_file(
     tag: str = DEMISTO_GIT_PRIMARY_BRANCH,
     return_content: bool = False,
 ):
+    """
+    Fetch a file from either local or remote repository.
+
+    For private repositories, tries local first then remote.
+    For public repositories, tries remote first then local.
+
+    Args:
+        full_file_path: Path to the file to fetch
+        tag: Git tag/branch to fetch from
+        return_content: Whether to return raw content or parsed file details
+
+    Returns:
+        File content (encoded bytes if return_content=True) or parsed file details (dict)
+    """
+    logger.debug(f"[get_local_remote_file] Fetching {full_file_path} (tag={tag})")
+
     repo_git_util = GitUtil()
-    git_path = repo_git_util.get_local_remote_file_path(full_file_path, tag)
-    file_content = repo_git_util.get_local_remote_file_content(git_path)
-    if return_content:
+
+    # Check if handling private repositories
+    is_handling_private_repo = os.getenv("DEMISTO_SDK_PRIVATE_REPO_MODE") == "true"
+
+    # Determine the order: private repos try local first, public repos try remote first
+    try_remote_first = not is_handling_private_repo
+
+    # Define the two sources to try in order
+    sources = [
+        ("remote", True) if try_remote_first else ("local", False),
+        ("local", False) if try_remote_first else ("remote", True),
+    ]
+
+    file_content = None
+    for source_name, from_remote in sources:
+        logger.debug(f"[get_local_remote_file] Trying {source_name}...")
+
+        git_path = repo_git_util.get_local_remote_file_path(
+            full_file_path, tag, from_remote=from_remote
+        )
+        logger.debug(
+            f"[get_local_remote_file] Git path ({source_name}) resolved to: {git_path}"
+        )
+
+        file_content = repo_git_util.get_local_remote_file_content(git_path)
```

Comment:
What happens if it doesn't exist on either remote or local? What is the behaviuor if we want to delete a file?